src/richie/apps/search/signals.py

```diff
@@ -165,3 +169,19 @@ def on_page_unpublished(sender, instance, language, **kwargs):
         transaction.on_commit(
             lambda: apply_es_action_to_page(instance, action, language)
         )
+
+
+# pylint: disable=unused-argument
+@receiver(post_obj_operation)
+def on_page_moved(sender, **kwargs):
+    """
+    When a page is moved, we may need to re-index all pages linked to objects of
+    the same kind. This applies to all pages linked to categories as they have the
+    *path* of the page in the ES index.
```

Comment:
It's not clear to me anymore why we need the path for categories and not for organizations or persons?
Do we really still need the path for categories?

Reply:
We need the paths to display the categories in the frontend with their parent-children hierarchy

Reply:
Ok. Yes it's true that it's useless for orgs and persons although theoretically we could have it.
Good thanks for the clarifications. I don't think we can avoid this without making ugly postgresql queries. So let's go.